xdebug.c

```diff
@@ -2050,9 +2072,15 @@ void xdebug_execute_internal(zend_execute_data *current_execute_data, zval *retu
 	}
 
 	/* Check for entry breakpoints */
-	if (xdebug_is_debug_connection_active_for_current_pid() && XG(breakpoints_allowed)) {
-		if (!handle_breakpoints(fse, XDEBUG_BREAKPOINT_TYPE_CALL)) {
-			xdebug_mark_debug_connection_not_active();
+	if(XG(breakpoints_allowed)){
+		block = 0;
+		if (xdebug_is_debug_connection_active_for_current_pid()) {
+			if (!handle_breakpoints(fse, XDEBUG_BREAKPOINT_TYPE_CALL, &block)) {
+				xdebug_mark_debug_connection_not_active();
+			}
+		}
+		if (xdebug_is_debug_connection_active_for_current_pid()) {
+			XG(context).handler->cmdloop(&(XG(context)), 1 TSRMLS_CC, block);
```

Comment:
As I have seen this same pattern multiple times now, this probably should be refactored, with an explanation as to why this is necessary.